diff -rupN gcc-4.2.4/config.sub ../gcc-4.2.4/config.sub
--- gcc-4.2.4/config.sub	2006-10-16 08:57:17.000000000 +0530
+++ ../gcc-4.2.4/config.sub	2009-02-14 08:35:44.921875000 +0530
@@ -1219,7 +1219,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -aceos*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
diff -rupN gcc-4.2.4/gcc/config/aceos.h ../gcc-4.2.4/gcc/config/aceos.h
--- gcc-4.2.4/gcc/config/aceos.h	1970-01-01 05:30:00.000000000 +0530
+++ ../gcc-4.2.4/gcc/config/aceos.h	2009-02-14 08:39:41.265625000 +0530
@@ -0,0 +1,11 @@
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()                \
+do {                                            \
+       builtin_define_std ("aceos");           \
+       builtin_define_std ("unix");            \
+       builtin_assert ("system=aceos");        \
+       builtin_assert ("system=unix");         \
+} while(0);
+
+#undef TARGET_VERSION
+#define TARGET_VERSION fprintf(stderr, " (i386 aceos)");
diff -rupN gcc-4.2.4/gcc/config.gcc ../gcc-4.2.4/gcc/config.gcc
--- gcc-4.2.4/gcc/config.gcc	2008-03-14 00:41:43.000000000 +0530
+++ ../gcc-4.2.4/gcc/config.gcc	2009-02-14 08:38:55.468750000 +0530
@@ -362,6 +362,12 @@ esac
 
 # Common parts for widely ported systems.
 case ${target} in
+*-*-aceos*)
+    extra_parts="crtbegin.o crtend.o"
+    gas=yes
+    gnu_ld=yes
+    default_use_cxa_atexit=yes
+    ;;
 *-*-darwin*)
   tm_file="${tm_file} darwin.h"
   case ${target} in
@@ -1032,6 +1038,11 @@ x86_64-*-darwin*)
 	tmake_file="t-darwin ${cpu_type}/t-darwin64 t-slibgcc-darwin"
 	tm_file="${tm_file} ${cpu_type}/darwin64.h"
 	;;
+i[34567]86-*-aceos*)
+       tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h aceos.h"
+       tmake_file="i386/t-i386elf t-svr4"
+       use_fixproto=yes
+       ;;
 i[34567]86-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h dbxelf.h elfos.h i386/i386elf.h"
 	tmake_file="i386/t-i386elf t-svr4"
